retrieval/discovery/marathon/conversion.go

```diff
@@ -0,0 +1,78 @@
+package marathon
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/prometheus/client_golang/model"
+	"github.com/prometheus/prometheus/config"
+)
+
+// ConvertAppsToTargetGroups takes an array of Marathon apps and converts them into target groups.
+func ConvertAppsToTargetGroups(clusterName string, useJobPrefix bool, apps *AppList) map[string]*config.TargetGroup {
+	tgroups := map[string]*config.TargetGroup{}
+	for _, a := range apps.Apps {
+		if isValidApp(&a) {
+			group := targetGroupFor(clusterName, useJobPrefix, &a)
+			tgroups[group.Source] = group
+		}
+	}
+	return tgroups
+}
+
+func targetGroupFor(clusterName string, useJobPrefix bool, app *App) *config.TargetGroup {
```

Comment:
I'd only use `For` if something is specified after that. Otherwise `createTargetGroup` or `newTargetGroup` are accurate.